File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {

            int[] query = queries[i];
            int s = query[0];
            int e = query[1];
            int k = query[2];

            answer[i] = -1;
            for (int j = s; j < e + 1; j++) {
                if (arr[j] > k) {
                    if (answer[i] == -1) {
                        answer[i] = arr[j];
                    } else {
                        answer[i] = Math.min(answer[i], arr[j]);
                    }
                }
            }
        }

        return answer;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.kt
import kotlin.math.min

internal class Solution {
    fun solution(arr: IntArray, queries: Array<IntArray>): IntArray {
        val answer = IntArray(queries.size)
        for (i in queries.indices) {
            val query = queries[i]
            val s = query[0]
            val e = query[1]
            val k = query[2]

            answer[i] = -1
            for (j in s until e + 1) {
                val digit = arr[j]
                if (k < digit) {
                    if (answer[i] == -1) {
                        answer[i] = digit
                    } else {
                        answer[i] = min(answer[i], digit)
                    }
                }
            }
        }

        return answer
    }
}